src/postamble.js

```diff
@@ -259,7 +261,41 @@ function run(args) {
   }
 
 #if STACK_OVERFLOW_CHECK
-  stackCheckInit();
+  if (!stackCheckInit.done) stackCheckInit();
+#endif
+
+#if RELOCATABLE
+  // Loading of dynamic libraries needs to happen on each thread, so we can't
+  // use the normal __ATPRERUN__ mechanism.
+#if MAIN_MODULE
+  preloadDylibs();
+#else
+#if RELOCATABLE
+  reportUndefinedSymbols();
+#endif
+#endif
+
+  // Loading dylibs can add run dependencies.
+  if (runDependencies > 0) {
+#if RUNTIME_LOGGING
+    err('run() called, but dependencies remain, so not running');
```

Comment:
I think this pattern (checking if deps exist, and showing this message if so) happens in another place - maybe worth a helper function?

Reply:
It only occurs twice now also the log message is now different,  Plus the primary part of this pattern is to return from the current function, which can't be done by a subroutine.